$MirOS: ports/print/cups/patches/patch-scheduler_cups-lpd_c,v 1.4 2007/03/15 19:22:40 bsiegert Exp $
--- scheduler/cups-lpd.c.orig	Sat Jul 12 00:48:26 2008
+++ scheduler/cups-lpd.c	Tue Oct 21 22:56:23 2008
@@ -1358,7 +1358,7 @@ remove_jobs(const char *dest,		/* I - De
 
     request = ippNewRequest(IPP_CANCEL_JOB);
 
-    sprintf(uri, "ipp://localhost/jobs/%d", id);
+    snprintf(uri, sizeof(uri), "ipp://localhost/jobs/%d", id);
     ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_URI, "job-uri", NULL, uri);
 
     ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_NAME,
